File: src/test/java/easy/FindTheDifferenceTest.java
package easy;

import static org.junit.jupiter.api.Assertions.*;

import java.util.stream.Stream;
import org.junit.jupiter.params.ParameterizedTest;
import org.junit.jupiter.params.provider.Arguments;
import org.junit.jupiter.params.provider.MethodSource;

public class FindTheDifferenceTest {
  FindTheDifference instance = new FindTheDifference();

  @ParameterizedTest
  @MethodSource("dataFindTheDifference")
  public void testFindTheDifference(String s, String t, char expected) {

    char result = instance.solve(s, t);

    assertEquals(expected, result);
  }

  public static Stream<Arguments> dataFindTheDifference() {
    return Stream.of(
        Arguments.of("abcd", "abcde", 'e'),
        Arguments.of("", "y", 'y'),
        Arguments.of("a", "aa", 'a'),
        Arguments.of("aabbcc", "aabbccd", 'd'));
  }
}

File: test/easyswift/FindTheDifferenceTest.swift
import Testing

@testable import easy

@Test(arguments: [
  ("abcd", "abcde", "e"),
  ("", "y", "y"),
  ("a", "aa", "a"),
  ("aabbcc", "aabbccd", "d"),
])
func findTheDifferenceTest(s: String, t: String, expected: Character?) {
  #expect(findTheDifference(s, t) == expected)
}
